content/blog/zephyr-tracing-and-profiling/index.md

```diff
@@ -0,0 +1,279 @@
+
+
+## Zephyr’s tracing tool
+
+In order to show how Zephyr’s native tracing system works, we’ll use the sample below and an ESP32-C6 to run it on a physical device:
+
+`samples/subsys/tracing`
+
+
+To enable Zephyr’s native tracing, a small number of options are available. For this example, which uses ESP32-C6’s USB JTAG port to output the logged data, we’ll set the following:
+
+```
+CONFIG_TRACING=y
+CONFIG_TRACING_CTF=y
+CONFIG_TRACING_BACKEND_UART=y
+CONFIG_TRACING_BUFFER_SIZE=4096
+```
+
+These settings activate the tracing feature, defines the data stored format as CTF (Common Trace Format) and configures the output medium (UART). As UART was selected as the medium to transfer data from the device to the host (PC), we’ll define the following settings in an overlay file within the sample:
+
+```
+/ {
+    chosen {
+        zephyr,tracing-uart = &usb_serial;
+    };
+};
+
+&usb_serial {
+    status = "okay";
+};
+```
+
+In order to build the sample we’ll use the following command:
+
+```shell
+west build -p -b esp32c6_devkitc samples/subsys/tracing/ -DCONF_FILE=prj_uart_ctf.conf -p
+```
+
+Once the sample is flashed, we can switch the USB cable to the USB port of the ESP32-C6 board and activate Zephyr’s python script to connect to the serial port and store the acquired data in a log file:
+
+```shell
+python3 scripts/tracing/trace_capture_uart.py -d /dev/ttyACM0 -b 115200 -o ../tracing/trace_0
+```
+
+The option -o indicates the path and file name for the tracing data, so any path and name can be used. In this example we’ll use an arbitrary folder called “tracing” just outside Zephyr's main folder.
+
+After some seconds of acquisition we can hit CTRL+C to stop recording and get back to the shell in the host PC.
+
+The tracing format CTF requires data typing to be decoded, so we’ll copy the metadata (tracing format description) file to the same folder:
+
+```shell
+cp subsys/tracing/ctf/tsdl/metadata ../tracing/
+```
+
+Finally, to parse the CTF data and show the actual trace:
+
+```shell
+babeltrace2 ../tracing/
+```
+
+The output is something like:
+
+<a href="img/img001.webp" target="_blank">
+    {{< figure
+        default=true
+        src="img/img001.webp"
+        alt=""
+        caption="Sample trace output"
+    >}}
+</a>
+
+This log shows the execution order and timing of the traced objects, allowing us to visualize how the software is executing in the acquired interval. This gives the possibility of running statistics on how the CPU resources are used and easily spot functions that are taking too long to execute, are being executed at wrong moments or more times than expected.

```

Comment:
last sentence sounds a bit off the whole phrasing:
`, are being executed at wrong moments or more times than expected.`